vm_eval.c

```diff
@@ -2508,9 +2527,16 @@ rb_current_realfilepath(void)
         if (path == eval_default_path) {
             return Qnil;
         }
-        else {
-            return path;
+
+        // [Feature #19755] implicit eval location is "(eval at #{__FILE__}:#{__LINE__})"
+        if (RSTRING_LEN(path) >= 9) {
```

Comment:
If it equals 9, the next `if` condition can't be true.
```suggestion
        if (RSTRING_LEN(path) > 9) {
```